tasks.py

```diff
@@ -1,12 +1,25 @@
+import os
 from pathlib import Path
+from shutil import which
 
 from invoke import task
 
 PKG_NAME = 'more_categories'
 PKG_PATH = Path(f'pelican/plugins/{PKG_NAME}')
-FLAKE8 = 'flake8'
-ISORT = 'isort'
-PYTEST = 'pytest'
+
+ACTIVE_VENV = os.environ.get("VIRTUAL_ENV", None)
+VENV_HOME = Path(os.environ.get("WORKON_HOME", "~/.local/share/virtualenvs"))
+VENV_PATH = Path(ACTIVE_VENV) if ACTIVE_VENV else (VENV_HOME / PKG_NAME)
+VENV = str(VENV_PATH.expanduser())
+
+
+TOOLS = ['poetry', 'pre-commit']
+FLAKE8 = which('flake8') if which('flake8') else (VENV / Path('bin') / 'flake8')
+ISORT = which('isort') if which('isort') else (VENV / Path('bin') / 'isort')
+PYTEST = which('pytest') if which('pytest') else (VENV / Path('bin') / 'pytest')
+PIP = which('pip') if which('pip') else (VENV / Path('bin') / 'pip')
+POETRY = which('poetry') if which('poetry') else (VENV / Path('bin') / 'poetry')
+PRECOMMIT = which('pre-commit') if which('pre-commit') else (VENV / Path('bin') / 'pre-commit')
```

Comment:
I think these ternary statements can all be simplified to:

```
FLAKE8 = which('flake8') or Path(f'{VENV}/bin/flake8')
```

Also, TOOLS should come at the end, and say

```
TOOLS = [POETRY, PRECOMMIT]
```

Reply:
If we change `TOOLS` to what you have, won't that break the loop that iterates over `TOOLS` in `tools()`? It feels counter-productive to do `TOOLS = [POETRY, PRECOMMIT]` since the point of `TOOLS` is to install `poetry` and `pre-commit` if they are not already installed.

Reply:
That's a good point! So if `PIP` is set to `Path(f'{VENV}/bin/pip')`, and we run `f'{PIP} install {tool}'`, will the tool be automatically installed in the virtual_env?

Reply:
From what I understand, it should.